Neos.ContentRepository/Classes/Domain/ContentStream/ContentStreamIdentifier.php

```diff
@@ -12,36 +10,44 @@
  * source code.
  */
 
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Domain\ContentStream;
+
 use Neos\Cache\CacheAwareInterface;
 use Neos\Flow\Utility\Algorithms;
 use Neos\Flow\Annotations as Flow;
 
 /**
  * The ContentStreamIdentifier is the identifier for a Content Stream, which is
  * a central concept in the Event-Sourced CR introduced with Neos 5.0.
- *
- * @Flow\Proxy(false)
  */
-final class ContentStreamIdentifier implements \JsonSerializable, CacheAwareInterface
+#[Flow\Proxy(false)]
+final class ContentStreamIdentifier implements \JsonSerializable, CacheAwareInterface, \Stringable
 {
     /**
-     * @var string
+     * @var array<string,self>
      */
-    private $value;
+    private static array $instances = [];
+
+    private function __construct(
+        private string $value
+    ) {
```

Comment:
nitpick: Why 3 lines for the constructor signature?

Reply:
With the constructor being the only place left where a property is declared, I prefer a separate line for that property to have all its attributes (type, readonly, scope) readily available.

Reply:
makes sense 👍

Reply:
Yes right, this makes sense!